networkx/algorithms/tests/test_distance_measures.py

```diff
@@ -97,6 +97,62 @@ def test_eccentricity_directed_weakly_connected(self):
             nx.eccentricity(DG)
 
 
+class TestWeightedDistance:
+    def setup_method(self):
+        G = nx.Graph()
+        G.add_edge(0, 1, weight=0.6)
+        G.add_edge(0, 2, weight=0.2)
+        G.add_edge(2, 3, weight=0.1)
+        G.add_edge(2, 4, weight=0.7)
+        G.add_edge(2, 5, weight=0.9)
+        G.add_edge(1, 5, weight=0.3)
+        self.G = G
+
+    def test_eccentricity(self):
+        assert nx.eccentricity(self.G, 1) == 1.5
+        e = nx.eccentricity(self.G)
+        assert e[1] == 1.5
+
+        e = nx.eccentricity(self.G, v=1)
+        assert e == 1.5
+
+        # This behavior changed in version 1.8 (ticket #739)
+        e = nx.eccentricity(self.G, v=[1, 1])
+        assert e[1] == 1.5
+        e = nx.eccentricity(self.G, v=[1, 2])
+        assert e[1] == 1.5
+
+    def test_diameter(self):
+        assert nx.diameter(self.G) == 1.6
+
+    def test_radius(self):
+        assert nx.radius(self.G) != 0.9
```

Comment:
OK... that's my mistake.  I was thinking we should check that we understand what is happening by testing that round-off occurs. But I agree that this is not the answer we care about. 

Should we make all the comparisons use the `pytest.approx` idiom? or only the ones where round-off messes up the comparison?

Reply:
> Should we make all the comparisons use the pytest.approx idiom? or only the ones where round-off messes up the comparison?

It's probably safest to use `pytest.approx` everywhere for testing "equality" with floating point values, but it's also not super critical. I doubt it will cause failures except *maybe* on systems that use 32-bit floats.

Reply:
Would it be useful to know (and test for) cases where round-off makes a difference. In these examples, 2 have round-off and the rest don't. Do we care if that changes? probably not. :}